facebook/react
# [RN] Remove __EXPERIMENTAL__ from enableOwnerStacks, enableUseDeferredValueInitialArg

Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #30322
* #30169
* #30072
* #30073
* #30071
* #30070

In www, the experimental versions get a .modern.js or .classic.js prefix
and get copied into the same folder. In RN, they don't seem to have
.modern.js and .classic.js versions so they end up getting the same
name.

@sebmarkbage's theory is that what happens is that they then override
the file that was already there. So depending on if experimental or
stable build finishes first you get a different version at the end.

It doesn't make sense to use `__EXPERIMENTAL__` for flags in native-fb
since there's no modern/classic split there. So that flag should just be
hardcoded to true or false and then it doesn't matter which one finishes
first.

We don't support experimental builds in OSS RN neither so the same thing
could happen with
[`enableOwnerStacks`](https://github.com/facebook/react/blob/5dcf3ca8d45a276a8b4cee0cedd234967661ca35/packages/shared/forks/ReactFeatureFlags.native-oss.js#L60).

You can see that the build errors in the previous PR but passes after
these flag changes.

## Test plan

1. Download gh build from https://github.com/facebook/react/actions/runs/9910037870/artifacts/1695683186
2. Download circle build from https://app.circleci.com/pipelines/github/facebook/react/56533/workflows/ff3b91bc-0af8-460a-a234-694531a50a44/jobs/950161/artifacts
3. Uncompress, remove size subdirectories due to ordering differences, compare circle and gh directories: https://gist.github.com/poteto/e46b37076262a71a86b7fa2d71d267fe. Note that the only difference is in the git hash (because of ghstack) and in the [variable names for useMemoCache in react-debug-tools](https://gist.github.com/poteto/e46b37076262a71a86b7fa2d71d267fe#file-diff-patch-L1025-L1090). Not 100% sure why these changed, but it's just variable names so it looks benign
